### PR TITLE
fix: partial workaround for high address support

### PR DESCRIPTION
**Pull request by chrisdickinson**
Some SDKs, like the js-sdk, store block index information in the top 16 bits of the 64-bit memory address. This is incompatible with the JS-PDK at present, since we truncate addresses to 32 bits. In advance of BigInt support in Javy/quickjs-wasm-sys, expose memory offset and lengths as JS Float values. This leans on the JS convention of storing (up to 53 bits of) integer data losslessly in an f64.

This gives us 5 bits of block index for the JS-SDK, enough room for *slaps roof of memory allocator* 32 allocations in this bad boy (before requiring an extism memory reset.)

QuickJS itself supports bigint, so it's a matter of looping that functionality out through quickjs-sys. (I haven't been able to find a technical reason this wasn't done initially, though that doesn't mean there aren't reasons!)

### CODE REVIEW

**Comment by nilslice** [comment]
> slaps roof of memory allocator

🤣

File: crates/core/src/globals.rs [hunk 1]
```diff
@@ -455,11 +455,16 @@ fn build_memory(context: &JSContextRef) -> anyhow::Result<JSValueRef> {
             if !ptr.is_number() {
                 bail!("Expected a pointer");
```
Comment (nilslice) [inline crates/core/src/globals.rs:456]: while you're in here....

do you think it makes sense to change some of these messages where `pointer` is used, to instead match the JS function names which use `offset`? 

e.g.
```patch
- bail!("Expected a pointer");
+ bail!("Expected an offset");
```

Comment (chrisdickinson) [reply crates/core/src/globals.rs]: Good call! Done in the latest commit.

File: crates/core/src/globals.rs [hunk 2]
```diff
@@ -442,47 +442,63 @@ fn build_memory(context: &JSContextRef) -> anyhow::Result<JSValueRef> {
             let data = data.as_bytes()?;
             let m = extism_pdk::Memory::from_bytes(data)?;
             let mut mem = HashMap::new();
-            let offset = JSValue::Int(m.offset() as i32);
-            let len = JSValue::Int(m.len() as i32);
+            let offset = JSValue::Float(m.offset() as f64);
```
Comment (bhelx) [inline crates/core/src/globals.rs:445]: Looks good and makes sense to me, but only really in the context of this PR. I think a single comment here about why we're packing the offset into a float may be helpful.

Comment (chrisdickinson) [reply crates/core/src/globals.rs:457]: Excellent point – added a note!

**Comment by chrisdickinson** [comment]
Thanks for the reviews, all!

**Pull request merged**